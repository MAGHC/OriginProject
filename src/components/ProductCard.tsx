import { GetProductTypeI } from '../type/GetProductType';
import { useNavigate } from 'react-router-dom';

interface PropsI {
  product: GetProductTypeI;
}

const ProductCard = ({ product, product: { id, img, title, price } }: PropsI) => {
  const navigate = useNavigate();
  return (
    <li
      className=" flex flex-col items-center"
      onClick={() => navigate(`/product/${id}`, { state: { product } })}
    >
      <img
        className=" rounded-[20px] cursor-pointer w-56 h-44 object-cover"
        alt={title}
        src={img}
      ></img>
      <div className=" w-40 md:w-44  font-light lg:w-48 xl:w-56 font-body  flex justify-between mx-auto   ">
        <h3 className=" truncate ... w-1/2">{title}</h3>
        <p className="  hidden lg:block  ">{`â‚©${price}`}</p>
      </div>
    </li>
  );
};

export default ProductCard;
