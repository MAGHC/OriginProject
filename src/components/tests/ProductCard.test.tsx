import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { Route, useLocation } from 'react-router-dom';

import ProductCard from '../ProductCard';

import { mockProduct } from '../../tests/data';
import { withRouter } from '../../tests/utils';

describe('ProductCard', () => {
  const { id, img, title, price, option, description, category } = mockProduct;

  it('snapshot test', () => {
    const component = renderer
      .create(
        withRouter(
          <Route path="/" element={<ProductCard product={mockProduct}></ProductCard>}></Route>,
        ),
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('product item render', async () => {
    render(
      withRouter(
        <Route path="/" element={<ProductCard product={mockProduct}></ProductCard>}></Route>,
      ),
    );
    const image = screen.getByAltText(title);
    expect(image).toHaveAttribute('src', img);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(`â‚©${price.toString()}`)).toBeInTheDocument();
  });

  it('naviagte to ProductDetail page when Clicked', () => {
    function TestComponent() {
      return <pre>{JSON.stringify(useLocation().state)}</pre>;
    }

    render(
      withRouter(
        <>
          <Route path="/" element={<ProductCard product={mockProduct} />} />
          <Route path={`/product/${id}`} element={<TestComponent />} />
        </>,
        ['/'],
      ),
    );

    const card = screen.getByRole('listitem');
    userEvent.click(card);

    expect(screen.getByText(JSON.stringify({ product: mockProduct }))).toBeInTheDocument();
  });
});

export {};
