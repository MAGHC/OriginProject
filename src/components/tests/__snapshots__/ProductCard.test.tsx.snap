// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot test 1`] = `
<li
  className=" flex flex-col items-center"
  onClick={[Function]}
>
  <img
    alt="test"
    className=" rounded-[20px] cursor-pointer w-56 h-44 object-cover"
    src=""
  />
  <div
    className=" w-40 md:w-44  font-light lg:w-48 xl:w-56 font-body  flex justify-between mx-auto   "
  >
    <h3
      className=" w-full lg:truncate ... lg:w-1/2"
    >
      test
    </h3>
    <p
      className="  hidden lg:block lg:w-1/2  "
    >
      â‚©13000
    </p>
  </div>
</li>
`;
